api/custom_auth/oauth/serializers.py

```diff
@@ -85,12 +83,10 @@ def _get_user(self, user_data: dict):
 
         if not existing_user:
             sign_up_type = self.validated_data.get("sign_up_type")
-            if not (
-                settings.ALLOW_REGISTRATION_WITHOUT_INVITE
-                or sign_up_type == SignUpType.INVITE_LINK.value
-                or Invite.objects.filter(email=email).exists()
-            ):
-                raise PermissionDenied(USER_REGISTRATION_WITHOUT_INVITE_ERROR_MESSAGE)
+            if not settings.ALLOW_REGISTRATION_WITHOUT_INVITE:
+                self._validate_registration_invite(
+                    {"email": email, **self.validated_data}
```

Comment:
I'd rather provide dependencies explicitly to avoid ambiguity
```suggestion
                    {"email": email, "sign_up_type": self.validated_data["sign_up_type"]}
```

Reply:
I've implemented this in a very similar way, but made it even more explicit.